README.md

```diff
@@ -29,3 +29,11 @@ protoc --doc_out="template=templates/tmpl.html:doc/" file.proto
 ```
 
 Would produce documentation using `templates/tmpl.html` inside the `doc/` output directory for `file.proto`.
+
+# proto_path issues
+
+You will quickly find that the generator fails if you run it with a `--proto_path` argument. This is because `protoc` does not pass it's command-line arguments onto generator programs (and `protoc-gen-doc` needs to know the `--proto_path` arguments in order to resolve symbols appropriately). To workaround this issue you must export the `$PROTO_PATH` environment variable, thus you can run `protoc` as just:
```

Comment:
Right, so it's a good question. Regarding command-line flags themselves, they aren't passed to you as a generator at all. There is a special set of flags (named `--foo_out="my-custom-flags --here"` where `foo` is derived from your `protoc-gen-foo` binary name), which _are passed_ to your binary but those don't include `--proto_path`.

I think what can be done is to buffer (or maybe stream) the files from the `protoc` generation request that is sent and then analyze the package statements afterwords through [CodeGeneratorRequest.ProtoFile](http://gotools.org/github.com/golang/protobuf/protoc-gen-go/plugin#plugin-pb-go-L33-L44) -> [FileDescriptorProto.Package](http://gotools.org/github.com/golang/protobuf/protoc-gen-go/descriptor#descriptor-pb-go-L258-L265).

The key to me finding that was this comment:
```
// FileDescriptorProtos for all files in files_to_generate and everything
// they import.  The files will appear in topological order, so each file
// appears before any file that imports it.
```

I think that's a far better approach than this one, and I already have plans to give it a shot  (I just sent this one as it _does_ work at the moment, until I can deduce the most proper fix for it; I should've written this in the original PR message).